src/Features/LanguageServer/Protocol/Handler/SpellCheck/DocumentSpellCheckHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.SpellCheck
+{
+    [Method(VSInternalMethods.TextDocumentSpellCheckableRangesName)]
+    internal class DocumentSpellCheckHandler : AbstractSpellCheckHandler<VSInternalDocumentSpellCheckableParams, VSInternalSpellCheckableRangeReport>
+    {
+        public override TextDocumentIdentifier? GetTextDocumentIdentifier(VSInternalDocumentSpellCheckableParams requestParams)
+            => requestParams.TextDocument;
+
+        protected override VSInternalSpellCheckableRangeReport CreateReport(TextDocumentIdentifier identifier, VSInternalSpellCheckableRange[]? ranges, string? resultId)
+            => new()
+            {
+                Ranges = ranges!,
```

Comment:
the ranges passed in can be null, see https://github.com/dotnet/roslyn/pull/59693/files#diff-10d38e1591e887d404de1bd900f3b35404ca5df88c31efc2492ee75a59a45072R211

The client package might need to be updated to allow null?

Reply:
yes.  i notified Hannah about this and she fixed this :)